### PR TITLE
Client connector refactoring

### PR DESCRIPTION
**Pull request by hendrikebbers**
Refactoring of the client connector API for future usage

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/358)

### CODE REVIEW

File: remoting/dolphin-remoting-client/src/main/java/org/opendolphin/core/client/comm/SimpleExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.opendolphin.core.client.comm;
+
+import java.util.concurrent.Executor;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Created by hendrikebbers on 15.02.17.
```
Comment (timo0) [inline remoting/dolphin-remoting-client/src/main/java/org/opendolphin/core/client/comm/SimpleExceptionHandler.java]: please remove header

File: remoting/dolphin-remoting-combined/src/test/groovy/org/opendolphin/core/comm/ClientConnectorPushTests.groovy [hunk 2]
```diff
@@ -41,43 +42,26 @@ class ClientConnectorPushTests extends Specification {
     // make sure the tests only count as ok if context.assertionsDone() has been reached
     protected void cleanup() {
         clientDolphin.sync { app.assertionsDone() }
-        assert app.done.await(4, TimeUnit.SECONDS) // max waiting time for async operations to have finished
+        assert app.done.await(4, TimeUnit.SECONDS) // max releaseNeeded time for async operations to have finished
```
Comment (timo0) [inline remoting/dolphin-remoting-combined/src/test/groovy/org/opendolphin/core/comm/ClientConnectorPushTests.groovy]: sure?

File: remoting/dolphin-remoting-shared/src/main/java/org/opendolphin/core/RemotingConstants.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package org.opendolphin.core;
+
+/**
+ * Created by hendrikebbers on 14.02.17.
```
Comment (timo0) [inline remoting/dolphin-remoting-shared/src/main/java/org/opendolphin/core/RemotingConstants.java]: remove headers

File: remoting/dolphin-remoting-combined/src/test/groovy/org/opendolphin/core/comm/ClientConnectorPushTests.groovy (hunk 2 above)
Comment (hendrikebbers) [inline remoting/dolphin-remoting-combined/src/test/groovy/org/opendolphin/core/comm/ClientConnectorPushTests.groovy]: lol :D this happened by refactoring the "waiting" value to "releaseNeeded" ;)

File: remoting/dolphin-remoting-client/src/main/java/org/opendolphin/core/client/comm/SimpleExceptionHandler.java (hunk 1 above)
Comment (hendrikebbers) [inline remoting/dolphin-remoting-client/src/main/java/org/opendolphin/core/client/comm/SimpleExceptionHandler.java]: ...

File: remoting/dolphin-remoting-shared/src/main/java/org/opendolphin/core/RemotingConstants.java (hunk 3 above)
Comment (hendrikebbers) [inline remoting/dolphin-remoting-shared/src/main/java/org/opendolphin/core/RemotingConstants.java]: ... :P

**Pull request merged**